cli/towerSysUtil.ml

```diff
@@ -103,5 +103,33 @@ module Dir = struct
       match err_opt with None -> make_exist ~perm path | Some err -> Some err
     ) None
 
+  let rec remove_dir path =
+    (* From http://pleac.sourceforge.net/pleac_ocaml/directories.html,
+ with some renaming for clarity *)
+    let rec process_tree f nodes =
+      Array.iter
+        (fun root ->
+         (match Unix.lstat root with
+          | {Unix.st_kind=Unix.S_DIR} ->
+             process_tree f
+                          (Array.map (Filename.concat root) (Sys.readdir root))
+          | _ -> ());
+         f root)
+        nodes in
+
+    let remove_node path =
+      match Unix.lstat path with
+      | {Unix.st_kind=Unix.S_DIR} ->
+         Printf.printf "rmdir %s\n%!" path;
+         Unix.rmdir path
+      | _ ->
+         Printf.printf "unlink %s\n%!" path;
+         Unix.unlink path
+    in
+    try
+      process_tree remove_node (Array.of_list [path])
+    with
+    | Unix.Unix_error (Unix.ENOENT, _, _) -> ()
```

Comment:
This can be caught if another process is operating on the same tree at the same time. I recommend handling the already missing case by explicitly checking for existence of `path` with `Sys.file_exists` and treating all `ENOENT` as errors.